config/config-api/src/com/thoughtworks/go/config/TemplatesConfig.java

```diff
@@ -100,15 +100,29 @@ public PipelineTemplateConfig templateByName(CaseInsensitiveString foo) {
 
     public boolean canViewAndEditTemplate(CaseInsensitiveString username, List<Role> roles) {
         for (PipelineTemplateConfig templateConfig : this) {
-            if (templateConfig.getAuthorization().getAdminsConfig().isAdmin(new AdminUser(username), roles)) {
+            if (canUserEditTemplate(templateConfig, username, roles)) {
                 return true;
             }
         }
         return false;
     }
 
-    public boolean canUserEditTemplate(String templateName, CaseInsensitiveString username, List<Role> roles) {
-        PipelineTemplateConfig templateConfig = this.templateByName(new CaseInsensitiveString(templateName));
-        return templateConfig.getAuthorization().getAdminsConfig().isAdmin(new AdminUser(username), roles);
+    public boolean canUserViewTemplates(CaseInsensitiveString username, List<Role> roles, boolean isGroupAdministrator) {
+        for (PipelineTemplateConfig templateConfig : this) {
+            if (hasViewAccessToTemplate(templateConfig, username, roles, isGroupAdministrator)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean canUserEditTemplate(PipelineTemplateConfig template, CaseInsensitiveString username, List<Role> roles) {
+        return template.getAuthorization().getAdminsConfig().isAdmin(new AdminUser(username), roles);
```

Comment:
Try pushing methods such as `getAuthorization().getViewConfig().isAdmin` and `getAuthorization().getAdminsConfig().isAdmin` to Authorization class